src/libs/actions/TaxRate.ts

```diff
@@ -0,0 +1,130 @@
+import type {TaxRate, TaxRates} from '@src/types/onyx';
+import type {PendingAction} from '@src/types/onyx/OnyxCommon';
+import type {OnyxData} from '@src/types/onyx/Request';
+
+/**
+ * Get tax value with percentage
+ */
+function getTaxValueWithPercentage(value: string): string {
+    return `${value}%`;
+}
+
+function covertTaxNameToID(name: string) {
+    return `id_${name.toUpperCase().replaceAll(' ', '_')}`;
+}
+
+/**
+ * Get new tax ID
+ */
+function getNextTaxID(name: string, taxRates?: TaxRates): string {
+    const newID = covertTaxNameToID(name);
+    if (!taxRates?.[newID]) {
+        return newID;
+    }
+
+    // If the tax ID already exists, we need to find a unique ID
+    let nextID = 1;
+    while (nextID < 100) {
+        const newNameCandidate = covertTaxNameToID(`${name}_${nextID}`);
+        if (!taxRates?.[newNameCandidate]) {
+            return newNameCandidate;
+        }
+        nextID++;
+    }
+    throw new Error('Could not find a unique tax ID');
+}
+
+function createWorkspaceTax(policyID: string, taxRate: TaxRate) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                ...taxRate,
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                                errors: null,
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                pendingAction: null,
+                                errors: null,
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxRate.code]: {
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                                errors: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.genericFailureMessage'),
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxFields: JSON.stringify({
```

Comment:
Why do we need to use `JSON.stringify` here? This is different from the [docs](https://docs.google.com/document/d/12FZycslrIE95QoKM8DZ37H_wLbH9w1jt3uA-a6BCqtc/edit)

Reply:
We have to stringify it to pass nested objects as a parameter. In the `Policy.ts` we have a few similar cases.

Reply:
It has to be stringified